.github/workflows/pr-title-check.yml

```diff
@@ -0,0 +1,26 @@
+name: Verify PR Title
+
+on:
+  pull_request:
+    types: [opened, edited, reopened]
+
+jobs:
+  verify-title:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
+
+      - name: Verify PR Title
+        id: verify
+        run: |
+          PREFIXES="query:|.*:"
+
+          if [[ ! "${{ github.event.pull_request.title }}" =~ ^(${PREFIXES}) ]]; then
+            echo "PR title does not follow the required format: 'prefix: description'"
+            echo "::error::PR title must start with one of the following prefixes: query:, .*:"
+            exit 1
+          else
+            echo "PR title is valid."
+          fi
```

Comment:
This is still wrong. Please reread the pull request template. The allowed prefix is not only `query:` and `.*:` (not that `.*` here isn't a regex it means "something that applies to all components).

Reply:
@douglascamata Okay, means "query" or some single word that applies to all components right? If it is then can you suggest what could be best possible word?

Reply:
The PR title, as the pull request template says, should start with the name of the Thanos component involved in the PR. We have a list of the components in the official docs (https://thanos.io/tip/components/). 

You can also check the history of PRs. You might see some that affected one or more components at the same time. For sure some that affected all of them too.

Reply:
PREFIXES="query:|compactor:|query-frontend:|receiver:|rule:|sidecar:|store:|tools:"
 
 @douglascamata Is this exactly correct?

Reply:
You're missing the `.*` (string) prefix, which has to be escaped to not be confused with the meaning that `.*` has in regexes.